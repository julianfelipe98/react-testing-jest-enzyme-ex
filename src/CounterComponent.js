// todo lo que comienze con use en el import del react significa que es un hook
import React, { useState } from "react";
import PropTypes from "prop-types";
const CounterComponent = ({ value = 10 }) => {
  const [counter, setCounter] = useState(value); //el use state me retorna un arreglo con dos valores
  // cuando se usa el setcounter le decimos a react que el estado de counter cambio por lo cual va a realizar el renderizado nuevamente
  const handleAdd = (e) => {
    // esta forma se usa cuando de pronto no tenemos el estado en el contexto y es enviado como parametro de algun otro lado la otra forma normal seria decirle en el st counter counter+1
    setCounter((c) => c + 1);
  };
  const handleSubstract = () => setCounter(counter - 1);

  const handleReset = () => setCounter(value);

  return (
    <>
      <h1>Counter Component</h1>
      <h2>{counter}</h2>

      <button onClick={handleAdd}>+1</button>
      <button onClick={handleReset}>reset</button>
      <button onClick={handleSubstract}>-1</button>
    </>
  );
};

CounterComponent.propTypes = {
  value: PropTypes.number,
};
export default CounterComponent;
