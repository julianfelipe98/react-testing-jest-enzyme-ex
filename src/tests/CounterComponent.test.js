import { shallow } from "enzyme";
import CounterComponent from "../CounterComponent";
// aun se puede simplificar codigo en estas pruebas
describe("CounterComponent Tests", () => {
  // podriamos solo declaralo pero se deja asi dado que si no perderiamos la ayuda del editor de texto
  let wrapper = shallow(<CounterComponent />);
  //   sentencia de cofigo que se ejecuta antes de ejecutar cada una de las pruebas , para el caso se instancia antes de cada prueba el wrapper
  beforeEach(() => {
    wrapper = shallow(<CounterComponent />);
  });

  test("It must to show the <CounterComponent/> component in the correct way", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("It must to show the 100 default value", () => {
    const defaultValue = 100;
    wrapper = shallow(<CounterComponent value={defaultValue} />);
    const textCounterComponent = parseInt(wrapper.find("h2").text());
    expect(textCounterComponent).toBe(defaultValue);
  });
  test('it must to increase with the "+1" button', () => {
    //   al simulate tambien le podemos enviar un segundo argumento seria los valores del evento
    const expectedResult = 11;
    wrapper.find("button").at(0).simulate("click");
    const textCounter = parseInt(wrapper.find("h2").text());
    expect(textCounter).toBe(expectedResult);
  });
  test('it must to decrease with the "-1" button', () => {
    //   al simulate tambien le podemos enviar un segundo argumento seria los valores del evento
    const expectedResult = 9;
    wrapper.find("button").at(2).simulate("click");
    const textCounter = parseInt(wrapper.find("h2").text());
    expect(textCounter).toBe(expectedResult);
  });
  test("It must to reset the CounterComponent default value", () => {
    const defaultValue = 100;
    const wrapper = shallow(<CounterComponent value={defaultValue} />);
    wrapper.find("button").at(2).simulate("click");
    wrapper.find("button").at(2).simulate("click");
    wrapper.find("button").at(1).simulate("click");
    const textCounter = parseInt(wrapper.find("h2").text());
    expect(textCounter).toBe(defaultValue);
  });
});
