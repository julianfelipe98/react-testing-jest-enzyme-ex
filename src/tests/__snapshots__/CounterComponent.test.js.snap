// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterComponent Tests It must to show the <CounterComponent/> component in the correct way 1`] = `
<Fragment>
  <h1>
    Counter Component
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
